Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/triangulate_faces.h

```diff
@@ -93,126 +92,167 @@ class Triangulate_modifier
     return fh->info().is_external;
   }
 
-  void triangulate_face(face_descriptor f, PM& pmesh)
+  bool triangulate_face(face_descriptor f, PM& pmesh)
   {
+
+
     typename Traits::Vector_3 normal =
       Polygon_mesh_processing::compute_face_normal(f, pmesh);
+    if(normal == typename Traits::Vector_3(0,0,0))
+      return false;
+    std::size_t original_size = CGAL::halfedges_around_face(halfedge(f, pmesh), pmesh).size();
+    if(original_size == 4)
+    {
+      halfedge_descriptor v0, v1, v2, v3;
+      v0 = halfedge(f, pmesh);
+      Point_ref p0 = _vpmap[target(v0, pmesh)];
+      v1 = next(v0, pmesh);
+      Point_ref p1 = _vpmap[target(v1, pmesh)];
+      v2 = next(v1, pmesh);
+      Point_ref p2 = _vpmap[target(v2, pmesh)];
+      v3 = next(v2, pmesh);
+      Point_ref p3 = _vpmap[target(v3, pmesh)];
 
-    P_traits cdt_traits(normal);
-    CDT cdt(cdt_traits);
+      double p1p3= CGAL::cross_product(p2-p1,p3-p2) * CGAL::cross_product(p0-p3,p1-p0);
```

Comment:
A small comment here explaining what is the logic would be great (to avoid reverse engineering)